Easydict/Feature/ViewController/Window/BaseQueryWindow/EZBaseQueryViewController.m

```diff
@@ -245,11 +245,31 @@ - (void)dealloc {
 - (void)handleServiceUpdate:(NSNotification *)notification {
     NSDictionary *userInfo = notification.userInfo;
     EZWindowType type = [userInfo[EZWindowTypeKey] integerValue];
+    NSString *serviceType = [notification.userInfo objectForKey:EZServiceTypeKey];
+    if ([serviceType length] != 0) {
+        [self updateService:serviceType];
+        return;
+    }
     if (type == self.windowType || !userInfo) {
         [self updateServices];
     }
 }
 
+- (void)updateService:(NSString *)serviceType {
+    NSMutableArray<EZQueryService *> *newServices = [[NSMutableArray alloc] init];
+    for (EZQueryService *service in self.services) {
+        if ([service serviceType] == serviceType) {
+            EZQueryService *item = [EZLocalStorage.shared service:serviceType];
+            [newServices addObject:item];
```

Comment:
When updating serve with this method, the new initialized serve doesn’t set `queryModel` and `windowType`, which will cause serve to fail to make a new query.

You need to improve `- (void)updateService:(NSString *)serviceType` by referring to `- (void)setupServices`.

Besides, it’s recommended that you refactor `- (void)setupServices` with the new `- (void)updateService:(NSString *)serviceType` method to decrease repeated code.

<img width="456" alt="image" src="https://github.com/tisfeng/Easydict/assets/25194972/f952448a-8932-4a2b-8ebc-aed20825f90e">

Reply:
Sure, changes have been updated, this issue should be fixed.